build/documentation/generate-typedocs.js

```diff
@@ -0,0 +1,198 @@
+ */
+
+var proc = dryRun ? exec(`node ${copyReadmeScriptPath} --dry-run`) : exec(`node ${copyReadmeScriptPath}`);
+
+proc.stdout.on('data', function(data) {
+    process.stdout.write(data);
+});
+
+proc.on('error', (err) => {
+    console.log(`\x1b[31m%s\x1b[0m`, err);
+});
+
+proc.on('close', function(code) {
+    console.log(`\x1b[32m%s\x1b[0m`, `copy-package-readme.js completed successfully.`);
+    execute();
+});
+
+/**
+ * Generate TypeDocs for each package
+ */
+function execute() {
+
+    if (dryRun) {
+        console.log(`In docs/en/packages/, this script would...`);
+    } else {
+        console.log(`Generating API documentation using TypeDoc...`);
+    }
+
+    const packages = path.resolve(rootDir, srcDir);
+
+    glob(packages, {realpath:true}, function(error, srcFiles) {
+       
+        srcFiles.forEach((srcPath) => {    
+            
+            var valid = true;
+
+            var packageName = srcPath
+                .split(path.sep)
+                .pop();
+
+            excludedPackages.forEach((excludedPackageName) => {
+                if (packageName == excludedPackageName) {
+                    console.log('\x1b[33m%s\x1b[0m', `...skip generating Typedoc API files for ${packageName}`);
+                    valid = false;
+                }
+            });
+
+            if(valid) {
+                dryRun ? console.log(`...generate Typedoc API files for ${packageName}`) : createAPIFiles(packageName, srcPath);
+                dryRun ? console.log(`...add API link to ${packageName}`) : addAPILinkToReadme(packageName);
+            }
+
+        });
+
+    });
+}
+
+/**
+ * Uses Typedoc process to generate API docs for a given package
+ * Stores them in the website/static folder 
+ */
+function createAPIFiles(packageName, srcPath) {
+    
+    var config = path.join(srcPath, '/tsconfig.json');
+    var output = path.join(destDir, packageName, 'api');
+    var file = path.join(srcPath, '/src/index.ts');
+
+    var typedocCmd = os.platform() == 'win32' ? 'typedoc.cmd' : 'typedoc';
+    var test = path.join('node_modules', '.bin', typedocCmd);
+
+    const typedoc = spawn(test, // the local typedoc installation is needed for the markdown theme to work
+        [
+            '--tsconfig', config,
+            '--out', output, file,
+            '--theme', "markdown"
+        ]);
+
+    typedoc.on('close', code => {
+        if (code) {
+            console.log('\x1b[31m%s\x1b[0m', `${packageName} - typedoc API docs not generated, error code: ${code}`);
+        } else {
+            console.log(`${packageName} - typedoc API docs generated`)
+            addHeaderToReadme(packageName);
+        }
+    });
+
+    // sometimes important errors will get sent to stdout for this process
+    // leaving this here because it's useful for debugging, but generates a lot
+    // of output, makes the terminal output harder to read so it's commented out
+
+    // typedoc.stdout.on('data', (data) => {
```

Comment:
I wonder if we want to add a future task to add an additional flag like `--verbose` that would then run this extra output useful for debugging?  Thoughts?

Reply:
I think it's a good idea, but I also think it'd have to be made clear that this option is for making it easier to debug the typedoc process when it generates an error. So it could maybe be mentioned 
like this:

```
if (verbose) { // use --verbose if you're having trouble debugging the typedoc process
    typedoc.stdout.on('data', (data) => {
        console.error(`typedoc stdout:\n${data}`);
    });
}
```

Reply:
This works for me. What is the likely hood of there being issues?  Is this a fairly delicate task?

Reply:
I think its should be alright, and fairly easy to implement. My only concern is that if there's a lot of places in the code where we're using this `if (verbose)` conditional, it would make the code a bit more difficult to read. But if it's just one or two spots, I think it should be fine.

Reply:
You have experience running and troubleshooting this script so I'll leave it up to you if you think this is useful or not and where it should be added. I'm also, ok with breaking this task out into a new issue and wiring it up later.

Reply:
Alright, I decided to leave it as is for now to make these changes easier to test. I'll add it in a consecutive PR.